test/tbb/test_concurrent_vector.cpp

```diff
@@ -692,14 +693,26 @@ TEST_CASE("swap with not always equal allocators"){
 // or fail with the assertion in debug mode.
 //! \brief \ref regression
 TEST_CASE("Testing vector in a highly concurrent environment") {
-    for (std::size_t i = 0; i < 10000; ++i) {
-        tbb::concurrent_vector<int> test_vec;
-
-        tbb::parallel_for(tbb::blocked_range<std::size_t>(0, 10000), [&] (const tbb::blocked_range<std::size_t>&) {
-            test_vec.grow_by(1);
-        }, tbb::static_partitioner{});
-
-        REQUIRE(test_vec.size() == utils::get_platform_max_threads());
+    std::uniform_int_distribution<unsigned> uniform_dist(1, 32); // grow by from 1 to 32 randomly
+    std::mt19937_64 gen(/*seed*/1); // Constructing with seed to have reproducible results
+    constexpr unsigned num_repeats = 10000, num_inserts = 256;
+    std::vector<int> grow_by_vals(num_inserts);
+
+    for (std::size_t i = 0; i < num_repeats; ++i) {
+        unsigned expected_size = 0;
+        std::generate(grow_by_vals.begin(), grow_by_vals.end(),
+                      [&gen, &uniform_dist, &expected_size]() {
+                          const unsigned random_value = uniform_dist(gen);
+                          expected_size += random_value;
+                          return random_value;
+                      });
+
+        tbb::concurrent_vector<double> test_vec;
+        tbb::parallel_for<unsigned>(0, num_inserts, [&] (unsigned j) {
+            test_vec.grow_by(grow_by_vals[j]);
+        });
```

Comment:
It seems like using `unsigned` is not important for the test and we can use `int` and improve readability.

Reply:
Well, this is subjective. Applied anyway though.